actors/user/src/main/java/org/sunbird/user/actors/AddressManagementActor.java

```diff
@@ -0,0 +1,123 @@
+package org.sunbird.user.actors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.actor.core.BaseActor;
+import org.sunbird.actor.router.ActorConfig;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.models.util.datasecurity.EncryptionService;
+import org.sunbird.common.request.Request;
+import org.sunbird.user.dao.AddressDao;
+import org.sunbird.user.dao.impl.AddressDaoImpl;
+
+@ActorConfig(
+  tasks = {"insertUserAddress", "updateUserAddress"},
+  asyncTasks = {"insertUserAddress", "updateUserAddress"}
+)
+public class AddressManagementActor extends BaseActor {
+
+  private EncryptionService encryptionService =
+      org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.getEncryptionServiceInstance(
+          null);
+  private AddressDao addressDao = AddressDaoImpl.getInstance();
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    String operation = request.getOperation();
+    switch (operation) {
+      case "insertUserAddress":
+        insertAddress(request);
+        break;
+
+      case "updateUserAddress":
+        updateAddress(request);
+        break;
+
+      default:
+        onReceiveUnsupportedOperation("AddressManagementActor");
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private void insertAddress(Request request) {
+    Map<String, Object> requestMap = request.getRequest();
+    List<Map<String, Object>> addressList =
+        (List<Map<String, Object>>) requestMap.get(JsonKey.ADDRESS);
+    Response response = new Response();
+    List<String> errMsgs = new ArrayList<>();
+    try {
+      String encUserId = encryptionService.encryptData((String) requestMap.get(JsonKey.ID));
+      String encCreatedById =
+          encryptionService.encryptData((String) requestMap.get(JsonKey.CREATED_BY));
+      for (int i = 0; i < addressList.size(); i++) {
+        Map<String, Object> address = addressList.get(i);
+        createAddress(encUserId, encCreatedById, address);
+      }
+    } catch (Exception e) {
+      errMsgs.add(e.getMessage());
```

Comment:
Add check for project common exception.
In case of generic exception, use generic error (e.g. Error in updating address {0}.)